projects/ADuCM3029_demo_ad7124_8PMDZ/src/app.c

```diff
@@ -0,0 +1,1806 @@
+ *                   called from the CLI.
+ *
+ * @return SUCCESS in case of success, negative error code otherwise.
+ */
+int32_t ad7124_8pmdz_help(struct ad7124_8pmdz_dev *dev, uint8_t *arg)
+{
+	int32_t ret;
+
+	ret = ad7124_8pmdz_help_prompt(dev, HELP_LONG_COMMAND);
+	if(ret != 0)
+		return ret;
+
+	ret = ad7124_8pmdz_help_general(dev, HELP_LONG_COMMAND);
+	if(ret != SUCCESS)
+		return ret;
+
+	ret = ad7124_8pmdz_help_adc(dev, HELP_LONG_COMMAND);
+	if(ret != SUCCESS)
+		return ret;
+
+	ret = ad7124_8pmdz_help_prompt(dev, HELP_SHORT_COMMAND);
+	if(ret != 0)
+		return ret;
+
+	ret = ad7124_8pmdz_help_general(dev, HELP_SHORT_COMMAND);
+	if(ret != SUCCESS)
+		return ret;
+
+	return ad7124_8pmdz_help_adc(dev, HELP_SHORT_COMMAND);
+}
+
+/**
+ * @brief CLI command to read and display a device register.
+ * @param [in] dev - Pointer to the application handler.
+ * @param [in] arg - The address of the register to be displayed.
+ * @return SUCCESS in case of success, FAILURE otherwise.
+ */
+int32_t ad7124_8pmdz_reg_read(struct ad7124_8pmdz_dev *dev, uint8_t *arg)
+{
+	uint8_t reg_addr, i;
+	struct ad7124_st_reg *reg_struct_ptr;
+	int32_t ret;
+	uint8_t buff[50];
+
+	reg_addr = strtol((char *)arg, NULL, 16);
+
+	for(i = 0; i < AD7124_REG_NO; i++) {
+		if(dev->ad7124_handler->regs[i].addr == reg_addr) {
+			reg_struct_ptr = &dev->ad7124_handler->regs[i];
+			ret = ad7124_read_register(dev->ad7124_handler, reg_struct_ptr);
+			break;
+		}
+	}
+	if(ret != SUCCESS)
+		return cli_write_string(dev->cli_handler,
+					    (uint8_t*)"SPI read error.\n");
+
+	if(i == AD7124_REG_NO)
+		return cli_write_string(dev->cli_handler,
+					    (uint8_t*)"Wrong register address.\n");
+
+	sprintf((char *)buff, "Register 0x%s value is: 0x%.6lx\n\r",
+			arg, dev->ad7124_handler->regs[i].value);
+
+	return cli_write_string(dev->cli_handler, buff);
+}
+
+/**
+ * @brief CLI command to write a device register.
+ * @param [in] dev - Pointer to the application handler.
+ * @param [in] arg - The address and new value of the register to be displayed.
+ * @return SUCCESS in case of success, FAILURE otherwise.
+ */
+int32_t ad7124_8pmdz_reg_write(struct ad7124_8pmdz_dev *dev, uint8_t *arg)
+{
+	uint8_t reg_addr, i;
+	uint8_t *reg_val_ptr;
+	struct ad7124_st_reg *reg_struct_ptr;
+	int32_t ret, reg_val;
+
+	reg_val_ptr = (uint8_t *)strchr((char *)arg, ' ');
+	*reg_val_ptr = 0;
+	reg_val_ptr++;
+
+	reg_addr = strtol((char *)arg, NULL, 16);
+
+	for(i = 0; i < AD7124_REG_NO; i++) {
+		if(dev->ad7124_handler->regs[i].addr == reg_addr) {
+			reg_struct_ptr = &dev->ad7124_handler->regs[i];
+			break;
+		}
+	}
+	if(i == AD7124_REG_NO)
+		return cli_write_string(dev->cli_handler,
+					    (uint8_t*)"Wrong register address.\n");
+	if(reg_struct_ptr->rw == AD7124_R)
+		return cli_write_string(dev->cli_handler,
+					    (uint8_t*)"Register is read only.\n");
+
+	reg_val = strtol((char *)reg_val_ptr, NULL, 16);
+
+	reg_struct_ptr->value = reg_val;

```

Comment:
by modifying this you modify the internal structure of ad7124_8pmdz_dev and an app shouldn't do that.

Reply:
I agree, but modifying the driver is out of scope for this project.